src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs

```diff
@@ -123,14 +126,21 @@ public int UpdateFrom4()
         // The UserName property rather than the NormalizedUserName is used as the ContentItem.Owner property matches the UserName.
         // New users will be created with a generated Id.
         // This code can be removed in a later version.
-        public async Task<int> UpdateFrom5Async()
+        public int UpdateFrom5()
         {
-            var users = await _session.Query<User>().ListAsync();
-            foreach (var user in users)
+            // Defer this until after the subsequent migrations have succeded as the schema has changed.
+            ShellScope.AddDeferredTask(async scope =>
             {
-                user.UserId = user.UserName;
-                _session.Save(user);
-            }
+                var session = scope.ServiceProvider.GetRequiredService<ISession>();
+                var users = await session.Query<User>().ListAsync();
+                foreach (var user in users)
+                {
+                    user.UserId = user.UserName;
+                    session.Save(user);
+                }
+
+                await session.SaveChangesAsync();
```

Comment:
Each deferred task is executed in its own scope, so normally `SaveChangesAsync()` is automatically called before disposing the scope. Can you retry it but without this line.

Reply:
Cool, I couldn't remember. Tested without, works fine